src/components/solvers/SolverPicker.tsx

```diff
@@ -96,21 +131,31 @@ export const SolverPicker = (props: SolverPickerProps) => {
     );
   };
 
+  let selectedSolver =
+    solverChoice.requestedSolverId !== undefined
+      ? solvers.find((s) => s.id === solverChoice.requestedSolverId)
+      : null;
+
   return (
     <Container>
       {loadingSolvers ? <Text>Loading solvers...</Text> : <SolverSelection />}
 
-      {solveRequest.requestedSolverId == undefined ? (
+      {selectedSolver?.authenticationOptions !== undefined &&
+      selectedSolver?.authenticationOptions !== null
+        ? Authentication(selectedSolver.authenticationOptions)
```

Comment:
Please use JSX (TSX) syntax

Reply:
I'm not sure what you mean

Reply:
Oh, you mean the short circuit way to have optional elements, gotcha. That's a neat trick.